src/ints/ems.cpp

```diff
@@ -630,15 +632,19 @@ static Bit8u MemoryRegion(void) {
 		dest_off=region.dest_offset&(MEM_PAGE_SIZE-1);
 		dest_remain=MEM_PAGE_SIZE-dest_off;
 	}
-	Bitu toread;
-	while (region.bytes>0) {
-		if (region.bytes>MEM_PAGE_SIZE) toread=MEM_PAGE_SIZE;
-		else toread=region.bytes;
+
+	uint8_t buf_src[MEM_PAGE_SIZE];
+	uint8_t buf_dest[MEM_PAGE_SIZE];
+
+	while (region.bytes > 0) {
+		const size_t toread = std::min<size_t>(region.bytes, MEM_PAGE_SIZE);
+
 		/* Read from the source */
 		if (!region.src_type) {
 			MEM_BlockRead(src_mem,buf_src,toread);
 		} else {
-			if (toread<src_remain) {
+			assert(toread <= MEM_PAGE_SIZE);
+			if (toread <= src_remain) {
 				MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,toread);
 			} else {
 				MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,src_remain);
```

Comment:
I was hoping we could keep collapsing all of these if / else blocks:
```
if (toread <= src_remain) {
	MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,toread);
} else {
	MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,src_remain);
```

into:
`MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src, std::min(toread, src_remain));

But then expanded the diff to see the remainder handed with a subsequent call.  Oh well :-)

Reply:
Yeah, I spent some time in the morning collapsing this into a separate function… The result was a function with 8 parameters and 1 template parameter… Turned out to not be more readable at all - it only shifted complexity from one place in code to another.

Half of the variables in this function should probably be collapsed into a second temporary MoveRegion variable or something… not sure. With the second attempt, I decided to limit scope only to lines directly relevant to the Coverity finding.